### PR TITLE
statement.d: improve Ddoc documentation

### PR DESCRIPTION
**Pull request by WalterBright**
Trying to raise the bar on this.

### CODE REVIEW

**Comment by JinShil** [comment]
Because of the way the documentation is presented online, I think the `Returns` documentation should begin with an upper case letter.

**Comment by JinShil** [comment]
"true" and "false" should be written as `$(D true)` and `$(D false)` when referring to code.

**Review by PetarKirov — APPROVED** [review]
LGTM

**Review by MetaLang — APPROVED** [review]

**Comment by MetaLang** [comment]
@wilzbach do you know why codecov is saying coverage went down?

File: src/ddmd/statement.d [hunk 1]
```diff
@@ -314,89 +353,119 @@ extern (C++) abstract class Statement : RootObject
     /*********************************
      * Flatten out the scope by presenting the statement
      * as an array of statements.
-     * Returns NULL if no flattening necessary.
+     * Params:
+     *    sc = context
+     * Returns:
+     *    The array of Statements, or null if no flattening necessary
```
Comment (JinShil) [inline src/ddmd/statement.d]: `null`

File: src/ddmd/statement.d [hunk 2]
```diff
@@ -66,6 +72,14 @@ extern (C++) Identifier fixupLabelName(Scope* sc, Identifier ident)
     return ident;
 }
 
+/*******************************************
+ * Check to see if statement is the innermost labeled statement.
+ * Params:
+ *      sc = context
+ *      statement = Statement to check
+ * Returns:
+ *      if true, then the LabelStatement, otherwise `null`
```
Comment (JinShil) [inline src/ddmd/statement.d]: true --> `true`

File: src/ddmd/statement.d [hunk 3]
```diff
@@ -172,19 +190,33 @@ extern (C++) abstract class Statement : RootObject
         return this;
     }
 
+    /****************************
+     * Determine if an enclosed `break` would apply to this
+     * statement, such as if it is a loop or switch statement.
+     * Returns:
+     *  `true` if it does
```
Comment (jacob-carlborg) [inline src/ddmd/statement.d]: The indentation of the `Returns:` sections are not consistent. This one is indented one space.

File: src/ddmd/statement.d [hunk 4]
```diff
@@ -172,19 +190,33 @@ extern (C++) abstract class Statement : RootObject
...
     bool hasContinue()
     {
         return false;
     }
 
-    /* ============================================== */
-    // true if statement uses exception handling
+    /**********************************
+     * Returns:
+     *   `true` if statement uses exception handling
```
Comment (jacob-carlborg) [inline src/ddmd/statement.d]: This one is indented two spaces.

File: src/ddmd/statement.d [hunk 5]
```diff
@@ -96,7 +110,10 @@ Expression checkAssignmentAsCondition(Expression e)
     return e;
 }
 
-/// Return a type identifier reference to 'object.Throwable'
+/**
+ * Returns:
+ *      TypeIdentifier corresponding to `object.Throwable`
```
Comment (jacob-carlborg) [inline src/ddmd/statement.d]: This one is indented five spaces.

File: src/ddmd/statement.d [hunk 6]
```diff
@@ -296,10 +332,13 @@ extern (C++) abstract class Statement : RootObject
...
      * a Statement.
-     * Output:
-     *      *sentry         code executed upon entry to the scope
-     *      *sexception     code executed upon exit from the scope via exception
-     *      *sfinally       code executed in finally block
+     * Params:
+     *    sc = context
+     *    sentry     = set to code executed upon entry to the scope
+     *    sexception = set to code executed upon exit from the scope via exception
+     *    sfinally   = set to code executed in finally block
```
Comment (jacob-carlborg) [inline src/ddmd/statement.d]: The parameters are not indented consistently with the other changes in this PR.

File: src/ddmd/statement.d [hunk 7]
```diff
@@ -107,6 +124,7 @@ TypeIdentifier getThrowable()
 
 
 /***********************************************************
+ * Specification: http://dlang.org/spec/statement.html
```
Comment (jacob-carlborg) [inline src/ddmd/statement.d]: Just a random idea: what about having the specification inline in Ddoc comments and render the web page based on the Ddoc comments.

File: src/ddmd/statement.d [hunk 8]
```diff
@@ -47,16 +47,22 @@ import ddmd.staticassert;
 import ddmd.tokens;
 import ddmd.visitor;
 
+/*****************************************
+ * CTFE requires FuncDeclaration::labtab for the interpretation.
+ * So fixing the label name inside in/out contracts is necessary
+ * for the uniqueness in labtab.
```
Comment (jacob-carlborg) [inline src/ddmd/statement.d]: * To me, this description sounds more like the reason why `fixupLabelName` is needed, not what it does
* Try to have a short summary [1]

[1] http://dlang.org/spec/ddoc.html#summary

File: src/ddmd/statement.d (hunk 7 above)
Comment (wilzbach) [inline src/ddmd/statement.d]: I proposed this a while ago:

https://github.com/dlang/dmd/pull/6877
https://github.com/dlang/dlang.org/pull/1683

Tl;Dr: @CyberShadow was worried about the added complexity to the dlang.org build process

Comment (jacob-carlborg) [inline src/ddmd/statement.d]: Right, I forgot about that PR.

File: src/ddmd/statement.d (hunk 6 above)
Comment (WalterBright) [inline src/ddmd/statement.d]: Doesn't matter.

Comment (jacob-carlborg) [inline src/ddmd/statement.d]: Not for the functionality. But it matters for consistency and that's important for readability.

File: src/ddmd/statement.d (hunk 8 above)
Comment (andralex) [reply src/ddmd/statement.d]: @jacob-carlborg could you please follow up on this with another PR? Thx!

**Review by andralex — APPROVED** [review]

**Comment by andralex** [comment]
Made changes prompted by the review myself - I encourage others to do the same. Thanks!

File: src/ddmd/statement.d (hunk 8 above)
Comment (jacob-carlborg) [reply src/ddmd/statement.d]: When did it become the norm that reviewers should fix issues in the PR? I will assume that when I create a PR next time 😉.

Comment (jacob-carlborg) [reply src/ddmd/statement.d]: I was not referring to Andrei fixing things in this PR. I was referring to his comment about me creating an other PR fixing what I commented on.

Comment (andralex) [reply src/ddmd/statement.d]: @jacob-carlborg you got a point, thanks for your response. @WalterBright's pull requests are "special" because as the creator and most prolific contributor to all aspects of the language, the allocation of his time is very important to D. So to the extent a pull request of his moves the ball forward and does not create technical debt, I'll allow them even if they could be made better. Though generally it's reasonable to ask him to take his pull requests to perfection, let us all improve on his work. Thanks!

Comment (jacob-carlborg) [reply src/ddmd/statement.d]: That's true, Walter's time is really valuable. At the same time I think you and Walter should lead by example. 🤔 it's not easy to balance these things.
